### PR TITLE
Add tests for script functions

### PR DESCRIPTION
**Pull request by LeoComandini**
- Fix bug in `scriptpubkey_is_multisig`
- Add missing tests for script functions
- Fix typos

### CODE REVIEW

File: src/test/test_script.py [hunk 1]
```diff
@@ -42,16 +58,229 @@ def test_scriptpubkey_p2pkh_from_bytes(self):
...
+        valid_args = [
+            (PK, PK_LEN, SIG, SIG_LEN, 0x01, out, out_len),
+            (PKU, PKU_LEN, SIG, SIG_LEN, 0x02, out, out_len),
+        ]
+        for args in valid_args:
+            ret = wally_scriptsig_p2pkh_from_sig(*args)
+            self.assertEqual(ret, (WALLY_OK, args[1] + args[3] + 9))
+
+    def test_scriptsig_multisig(self):
+        """Tests for creating mulisig scriptsig"""
```
Comment (jkauffman1) [inline src/test/test_script.py]: nit: typo 'mulisig'

File: src/test/test_script.py [hunk 2]
```diff
@@ -72,16 +301,17 @@ def test_script_push_from_bytes(self):
...
+                       ('00' * 50, SCRIPT_HASH160, '0014f71015b29ff2583558877ed16a59e4f8f451daa3'),
+                       ('00' * 50, SCRIPT_SHA256, '0020cc2786e1f9910a9d811400edcddaf7075195f7a16b216dcbefba3bc7c4f2ae51')]
 
         out, out_len = make_cbuffer('00' * 100)
-        for data, flags in valid_cases:
+        for data, flags, exp_program in valid_cases:
             in_, in_len = make_cbuffer(data)
             ret, written = wally_witness_program_from_bytes(in_, in_len, flags, out, out_len)
             self.assertEqual(ret, WALLY_OK)
+            self.assertEqual(out[:(len(exp_program)//2)], unhexlify(exp_program))
```
Comment (jkauffman1) [inline src/test/test_script.py]: nit: We could possibly compare to out[:written] here which would also test that written is correct

Comment (jgriffiths) [reply src/test/test_script.py]: Updated while merging, thanks

**Comment by jgriffiths** [comment]
Merged, thanks!

**Pull request closed without merging**